tests/regression_tests/unstructured_mesh/test.py

```diff
@@ -0,0 +1,270 @@
+        water_max_z = openmc.ZPlane(z0=bounds[2],
+                                    name="maximum z",
+                                    boundary_type='vacuum')
+
+        water_cell = openmc.Cell(name="water")
+        water_cell.region = (-clad_min_x | +clad_max_x | \
+                             -clad_min_y | +clad_max_y | \
+                             -clad_min_z | +clad_max_z) & \
+                             (+water_min_x & -water_max_x & \
+                              +water_min_y & -water_max_y & \
+                              +water_min_z & -water_max_z)
+        water_cell.fill = water_mat
+
+        # create a containing universe
+        root_univ = openmc.Universe()
+        root_univ.add_cells([fuel_cell, clad_cell, water_cell])
+
+        geom = openmc.Geometry(root=root_univ)
+
+        geom.export_to_xml()
+
+        ### Tallies ###
+
+        # create meshes
+        regular_mesh = openmc.RegularMesh()
+        regular_mesh.dimension = (10, 10, 10)
+        regular_mesh.lower_left = (-10.0, -10.0, -10.0)
+        regular_mesh.upper_right = (10.0, 10.0, 10.0)
+
+        regular_mesh_filter = openmc.MeshFilter(mesh=regular_mesh)
+
+        uscd_mesh = openmc.UnstructuredMesh()
+        uscd_mesh.filename = self.mesh_filename
+        uscd_mesh.mesh_lib = 'moab'
+        uscd_filter = openmc.MeshFilter(mesh=uscd_mesh)
+
+        # create tallies
+        tallies = openmc.Tallies()
+
+        regular_mesh_tally = openmc.Tally(name="regular mesh tally")
+        regular_mesh_tally.filters = [regular_mesh_filter]
+        regular_mesh_tally.scores = ['flux']
+        regular_mesh_tally.estimator = self.estimator
+        tallies.append(regular_mesh_tally)
+
+        uscd_tally = openmc.Tally(name="unstructured mesh tally")
+        uscd_tally.filters = [uscd_filter]
+        uscd_tally.scores = ['flux']
+        uscd_tally.estimator = self.estimator
+        tallies.append(uscd_tally)
+
+        tallies.export_to_xml()
+
+        ### Settings ###
+        settings = openmc.Settings()
+        settings.run_mode = 'fixed source'
+        settings.particles = 100
+        settings.batches = 10
+
+        # source setup
+        r = openmc.stats.Uniform(a=0.0, b=0.0)
+        theta = openmc.stats.Discrete(x=[0.0], p=[1.0])
+        phi = openmc.stats.Discrete(x=[0.0], p=[1.0])
+        origin = (0.0, 0.0, 0.0)
+
+        space = openmc.stats.SphericalIndependent(r=r,
+                                                  theta=theta,
+                                                  phi=phi,
+                                                  origin=origin)
+
+        angle = openmc.stats.Monodirectional((-1.0, 0.0, 0.0))
+
+        energy = openmc.stats.Discrete(x=[15.e+06], p=[1.0])
+
+        source = openmc.Source(space=space, energy=energy, angle=angle)
+
+        settings.source = source
+
+        settings.export_to_xml()
+
+    def _compare_inputs(self):
+        pass
```

Comment:
I added this because the xml's are going to differ depending on the test being run as a result of using `pytest`'s parameterize here. The unstructured mesh filename and tally estimator (maybe other values too?) in the `tally.xml` will be updated depending on the set of options passed to the `UnstructuredMeshTest` class. 

There's probably a better solution to this though -- bypassing a standard test function like isn't great it's true.

Reply:
In my last commit here, I've submitted a possible solution for this which allows the `PyAPITestHarness` to take in a filename for the input check which it will use to both update and check the test results.